src/haddock/modules/refinement/emref/cns/read_data.cns

```diff
@@ -23,22 +22,30 @@ fileexist $filenam0 end
 if ($result eq false) then
     evaluate ($filenam0 = $ambig_fname)
 end if
-fileexist $filenam0 end
-if ($result eq true) then
-    noe class ambi @@$filenam0 end
+
+if ($filenam0 # "") then
+    fileexist $filenam0 end
+    if ($result eq true) then
+        noe class ambi @@$filenam0 end
+    end if
 end if
+
 if ($Data.randremoval eq true) then
     noe part $Data.npart end
 end if
 
-fileexist $unambig_fname end
-if ($result eq true) then
-    noe class dist @@$unambig_fname end
+if ($unambig_fname # "") then
+    fileexist $unambig_fname end
+    if ($result eq true) then
+        noe class dist @@$unambig_fname end
+    end if
 end if
 
-fileexist $hbond_fname end
-if ($result eq true) then
-    noe @@$hbond_fname end
+if ($hbond_fname # "") then
+    fileexist $hbond_fname end
+    if ($result eq true) then
+        noe @@$hbond_fname end
```

Comment:
why not `noe class dist @@$hbond_fname end` in this case ?

Reply:
Good point! In haddock2.X the unambiguous file was always existing, even if empty. In haddock3 this is no longer the case. So the hbond will not inherit the dist class in case unambig_fname is not defined.

Reply:
actually I will open a new issue for this